src/Crypto/KeyPair.php

```diff
@@ -21,17 +21,13 @@ class KeyPair
      *
      * If the public key is left empty, it will be generated on demand from the private key.
      *
+     * WARNING: Generating a public key from the private key is a slow process. If available, please provide both keys.
+     *
      * @param PrivateKey $private
      * @param PublicKey|null $public
-     *
-     * @throws InvalidArgumentException When private and public key do not match.
      */
     public function __construct(PrivateKey $private, PublicKey $public = null)
     {
-        if (!$this->isValidKeyPair($private, $public)) {
-            throw new InvalidArgumentException('KeyPair could not be created: private and public key do not match.');
```

Comment:
This turned out to be fairly slow, and as it was 'only' an assertion, i removed it until i can think of a better way to check if the public key belongs the the private one.

Reply:
i think we might have to accept there isn't a fast way to do this.

Question then remains, do we check anyway or go for speed?